# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose Lane 2: Refresh / Content Opportunity Scoring. I want to identify which content pages should be reviewed first because editors have limited time and cannot inspect every page manually.

The project will rank pages using measured signals and an editor can then review each highly ranked page and decide whether to refresh, expand, protect, prune, or monitor it.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

1. Decision: Which pages should editors prioritize for review?

2. Actor: A content editor reviews the highest-ranked pages and chooses an appropriate action.

3. Wrong recommendation cost: A false positive wastes editorial time and could lead to unnecessary changes. A false negative means overlooking a page that may deserve attention.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

In [2]:
not_updated_90d = df["days_since_last_update"] > 90
visible = df["impressions_90d"] >= 100
older_and_visible = not_updated_90d & visible
print(
    f"{older_and_visible.sum():,} pages ({older_and_visible.mean():.1%} of all {len(df)} pages) "
    "have not been updated in more than 90 days "
    "and received at least 100 impressions."
)

8,119 pages (27.1% of all 30000 pages) have not been updated in more than 90 days and received at least 100 impressions.


In [3]:
visible = df["impressions_90d"] >= 100
older = df["days_since_last_update"] > 90
declining = df["trend_direction"].str.lower().eq("down")

declining_rate_recent_visible = declining[visible & ~older].mean()
declining_rate_older_visible = declining[visible & older].mean()
difference = declining_rate_older_visible - declining_rate_recent_visible

print(
    f"Downward-trending rate among visible pages updated within the last 90 days: "
    f"{declining_rate_recent_visible:.1%}"
)
print(
    f"Downward-trending rate among visible pages older than 90 days: "
    f"{declining_rate_older_visible:.1%}"
)
print(f"Observed difference: {difference:.1%}")

Downward-trending rate among visible pages updated within the last 90 days: 58.3%
Downward-trending rate among visible pages older than 90 days: 62.3%
Observed difference: 4.0%


Of the 30,000 pages, 8,119 (27.1%) received at least 100 impressions and had not been updated in more than 90 days, creating a sizable pool for editorial review. Among visible pages, 62.3% of those not updated in more than 90 days had a measured downward trend, compared with 58.3% of those updated within 90 days, a difference of 4.0 percentage points. This observed association suggests that time since the last update may be useful as one signal for prioritizing pages for review, but it does not show that page age caused the decline or that refreshing a page would improve traffic.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I will use observed content and performance signals to rank pages for review and suggest whether each page should be refreshed, expanded, protected, pruned, or monitored. High rankings will be supported by reason codes and a confidence level, and I will assess whether the top of the queue is more useful than a simple baseline. The results may support prioritization, but they will not show that an edit caused or will cause improvement, predict Google’s decisions, or identify Google ranking factors.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.